jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java

```diff
@@ -213,15 +213,18 @@ public void testFlushWhenQueueFull() {
   public void testFlushUpdatesQueueLength() throws Exception {
     // change sender to blocking mode
     sender.permitAppend(0);
-    RemoteReporter remoteReporter = (RemoteReporter) reporter;
-    remoteReporter.flush();
+
+    await("flush() execution via TimerTask").atMost(5, TimeUnit.SECONDS).until(
+        () -> metricsReporter.gauges.get("jaeger.reporter-queue") != null
+    );
```

Comment:
I have a couple issues with this. First, because the flush interval is 1s, this test can block for up to a second. It's better to create a new instance of the remote reporter just for this test with a shorter interval.

Second, why wait for timer flush in the first place? That doesn't seem to be the intention of this test. I could see the test could be simpler:
  * use reporter with large flush interval to ensure timer flush won't happen in the test
  * block the sender
  * retrieve the current value of the queue length metric (null == 0)
  * append N spans (maybe N=3)
  * call `flush()` explicitly
  * verify that the gauge has a value of N or N-1

Is there a race condition in such algorithm?